Add optional replacement to `ForbiddenDirectives`

TimoBakx opened the pull request:
Closes #1900

OskarStark commented:
After this PR is merged, I would please a new version, then we need to wait ~10 minutes and then a PR against `symfony/symofny-docs:@6.4` would be the next step, raising the version in the GithubActions workflow file + adjusting the `.doctor-rst.yaml` file accordingly.

wouterj commented on src/Rule/ForbiddenDirectives.php:
```diff
@@ -29,15 +30,24 @@ class ForbiddenDirectives extends AbstractRule implements Configurable, LineCont
 
     /**
-     * @var array<string>
+     * @var array<array[directive<string>, replacement<?string>]>
```
```suggestion
     * @var array<array{directive: string, replacement: ?string}>
```

OskarStark replied:
I don't think so @wouterj

Can you maybe add a test for the configuration?

TimoBakx commented:
Updated the code so it takes an array of replacements. Refactored the tests a bit because there are a lot more variations now. Also added tests for configuration failures.

OskarStark commented:
Thanks @TimoBakx 

You may want to send a PR to `symfony-docs` like mentioned before?

TimoBakx commented:
Yes, I will do that.
